### PR TITLE
chore(core): Update matrix: add pypy3, drop py27

### PR DESCRIPTION
**Pull request by ceache**
## Why is this needed?

By dropping py2.7 and pypy, we drop the last of the py2.x syntax from our tests matrix

## Proposed Changes

  - Tests are now only run only on py37, py38 and pypy3.
  - All py versions run all tests with gevent and sasl.
  - Bumped Zookeeper to 3.5.8.
  - Update .travis.yml definition per warnings, only syntax refresh.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -11,57 +11,33 @@ cache:
...
-  - python: '3.7'
-    env: ZOOKEEPER_VERSION=3.5.7 ZOOKEEPER_PREFIX="apache-" ZOOKEEPER_SUFFIX="-bin" ZOOKEEPER_LIB="lib" TOX_VENV=py37-sasl
-  - python: '3.8'
-    env: ZOOKEEPER_VERSION=3.4.14 TOX_VENV=py38
-  - python: '3.8'
-    env: ZOOKEEPER_VERSION=3.4.14 TOX_VENV=py38-sasl
+    env: ZOOKEEPER_VERSION=3.5.8 ZOOKEEPER_PREFIX="apache-" ZOOKEEPER_SUFFIX="-bin" ZOOKEEPER_LIB="lib" TOX_VENV=py37-gevent-sasl
   - python: '3.8'
-    env: ZOOKEEPER_VERSION=3.5.7 ZOOKEEPER_PREFIX="apache-" ZOOKEEPER_SUFFIX="-bin" ZOOKEEPER_LIB="lib" TOX_VENV=py38 DEPLOY=true
+    env: ZOOKEEPER_VERSION=3.4.14 TOX_VENV=py38-gevent
```
Comment (StephenSorriaux) [inline .travis.yml:21]: Should SASL be tested also with CPython 3.8?

File: .travis.yml [hunk 2]
```diff
@@ -11,57 +11,33 @@ cache:
...
-    env: ZOOKEEPER_VERSION=3.5.7 ZOOKEEPER_PREFIX="apache-" ZOOKEEPER_SUFFIX="-bin" ZOOKEEPER_LIB="lib" TOX_VENV=py38-sasl
-  - python: pypy
-    env: ZOOKEEPER_VERSION=3.4.14 TOX_VENV=pypy
-  - python: 'pypy'
-    env: ZOOKEEPER_VERSION=3.5.7 ZOOKEEPER_PREFIX="apache-" ZOOKEEPER_SUFFIX="-bin" ZOOKEEPER_LIB="lib" TOX_VENV=pypy
+    env: ZOOKEEPER_VERSION=3.5.8 ZOOKEEPER_PREFIX="apache-" ZOOKEEPER_SUFFIX="-bin" ZOOKEEPER_LIB="lib" TOX_VENV=py38-gevent-sasl DEPLOY=true
+  - python: 'pypy3'
+    env: ZOOKEEPER_VERSION=3.4.14 TOX_VENV=pypy3-gevent-sasl
+  - python: 'pypy3'
+    env: ZOOKEEPER_VERSION=3.5.6 ZOOKEEPER_PREFIX="apache-" ZOOKEEPER_SUFFIX="-bin" ZOOKEEPER_LIB="lib" TOX_VENV=pypy3-gevent-sasl
```
Comment (StephenSorriaux) [inline .travis.yml:27]: Should pypy3 be tested on ZK 3.5.8?

File: .travis.yml (hunk 1 above)
Comment (ceache) [reply .travis.yml]: fixed, thanks!

File: .travis.yml (hunk 2 above)
Comment (ceache) [reply .travis.yml]: fixed

**Comment by ceache** [comment]
@StephenSorriaux addressed your comments and also re-added eventlet

**Pull request merged**